modules/acquia_cms_common/acquia_cms_common.module

```diff
@@ -456,3 +463,18 @@ function acquia_cms_common_preprocess_maintenance_page(array &$variables) {
   $acquia_cms_path = \Drupal::service('extension.list.module')->getPath('acquia_cms_common');
   $variables['install_page_logo_path'] = '/' . $acquia_cms_path . '/acquia_cms.png';
 }
+
+/**
+ * Implements hook_module_implements_alter().
+ */
+function acquia_cms_common_module_implements_alter(array &$implementations, string $hook): void {
+  if (\Drupal::moduleHandler()->moduleExists('acquia_connector') &&
+  ($hook === 'modules_installed' || $hook === 'modules_uninstalled')) {
+    // Check if telemetry not opted.
+    if (!\Drupal::state()->get('acquia_connector.telemetry.opted')) {
```

Comment:
@apathak18 Let's pass the second argument to `state()->get()` as TRUE which opt in the user, if no value exist.
```suggestion
    if (!\Drupal::state()->get('acquia_connector.telemetry.opted', TRUE)) {
```
Let's do the same in eventSubscriber as well (unless you see any issue).